### PR TITLE
osd: fix be_deep_scrub() can't catch omap error

### PR DESCRIPTION
**Pull request by xiexingguo**
be_deep_scrub() will scan the omap of the specified object also.
However, this can not work well currently because iter->status()
does not return -EIO on error, it returns -1 instead.

Signed-off-by: xie xingguo <xie.xingguo@zte.com.cn>

### CODE REVIEW

File: src/osd/ReplicatedBackend.cc [hunk 1]
```diff
@@ -806,6 +806,13 @@ void ReplicatedBackend::be_deep_scrub(
   assert(iter);
   uint64_t keys_scanned = 0;
   for (iter->seek_to_first(); iter->valid() ; iter->next(false)) {
+    if (iter->status() < 0) {
```
Comment (tchaikov) [inline src/osd/ReplicatedBackend.cc]: can we change this way?
```c++
for (iter->seek_to_first(); !iter->status() && iter->valid(); iter->next(false)) {
```

and move the check out of the for loop?

**Comment by xiexingguo** [comment]
Done, thanks for the review:)

**Comment by liewegas** [comment]
@tchaikov ping

**Comment by tchaikov** [comment]
lgtm

**Pull request merged**